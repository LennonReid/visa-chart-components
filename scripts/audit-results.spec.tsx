/**
 * Copyright (c) 2021 Visa, Inc.
 *
 * This source code is licensed under the MIT license
 * https://github.com/visa/visa-chart-components/blob/master/LICENSE
 *
 **/
import combinedAuditResults from '../coverage/yarn-audit-report.json';

describe('<yarn-audit-results>', () => {
  it('audit results should match what we had before, and be clean', async () => {
    expect(combinedAuditResults).toMatchSnapshot();
  });
});
