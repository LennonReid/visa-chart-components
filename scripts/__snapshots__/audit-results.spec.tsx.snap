// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<yarn-audit-results> audit results should match what we had before, and be clean 1`] = `
Object {
  "alluvial-diagram": Array [
    Object {
      "data": Object {
        "dependencies": 109,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 109,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "bar-chart": Array [
    Object {
      "data": Object {
        "dependencies": 108,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 108,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "bivariate-mapbox-map": Array [
    Object {
      "data": Object {
        "dependencies": 172,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 172,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "charts": Array [
    Object {
      "data": Object {
        "dependencies": 142,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 142,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "charts-angular": Array [
    Object {
      "data": Object {
        "advisory": Object {
          "access": "public",
          "created": "2022-05-27T00:01:08.000Z",
          "cves": Array [
            "CVE-2021-4231",
          ],
          "cvss": Object {
            "score": 5.4,
            "vectorString": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:C/C:L/I:L/A:N",
          },
          "cwe": Array [
            "CWE-79",
          ],
          "deleted": null,
          "findings": Array [
            Object {
              "paths": Array [
                "codelyzer>@angular/core",
              ],
              "version": "9.0.0",
            },
          ],
          "found_by": null,
          "github_advisory_id": "GHSA-c75v-2vq8-878f",
          "id": 1070482,
          "metadata": null,
          "module_name": "@angular/core",
          "npm_advisory_id": null,
          "overview": "A vulnerability was found in Angular up to 11.0.4/11.1.0-next.2. It has been classified as problematic. Affected is the handling of comments. The manipulation leads to cross site scripting. It is possible to launch the attack remotely but it might require an authentication first. Upgrading to version 11.0.5 and 11.1.0-next.3 is able to address this issue. The name of the patch is ba8da742e3b243e8f43d4c63aa842b44e14f2b09. It is recommended to upgrade the affected component.",
          "patched_versions": ">=11.0.5",
          "recommendation": "Upgrade to version 11.0.5 or later",
          "references": "- https://nvd.nist.gov/vuln/detail/CVE-2021-4231
- https://github.com/angular/angular/issues/40136
- https://github.com/angular/angular/commit/ba8da742e3b243e8f43d4c63aa842b44e14f2b09
- https://security.snyk.io/vuln/SNYK-JS-ANGULARCORE-1070902
- https://vuldb.com/?id.181356
- https://github.com/advisories/GHSA-c75v-2vq8-878f",
          "reported_by": null,
          "severity": "moderate",
          "title": "Cross site scripting in Angular",
          "updated": "2022-06-08T20:06:50.000Z",
          "url": "https://github.com/advisories/GHSA-c75v-2vq8-878f",
          "vulnerable_versions": "<11.0.5",
        },
        "resolution": Object {
          "bundled": false,
          "dev": false,
          "id": 1070482,
          "optional": false,
          "path": "codelyzer>@angular/core",
        },
      },
      "type": "auditAdvisory",
    },
    Object {
      "data": Object {
        "dependencies": 1248,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 1248,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 1,
        },
      },
      "type": "auditSummary",
    },
  ],
  "charts-react": Array [
    Object {
      "data": Object {
        "dependencies": 57,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 57,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "circle-packing": Array [
    Object {
      "data": Object {
        "dependencies": 109,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 109,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "clustered-bar-chart": Array [
    Object {
      "data": Object {
        "dependencies": 109,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 109,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "data-table": Array [
    Object {
      "data": Object {
        "dependencies": 102,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 102,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "dumbbell-plot": Array [
    Object {
      "data": Object {
        "dependencies": 108,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 108,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "heat-map": Array [
    Object {
      "data": Object {
        "dependencies": 108,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 108,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "js": Array [
    Object {
      "data": Object {
        "dependencies": 638,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 638,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "keyboard-instructions": Array [
    Object {
      "data": "No lockfile found.",
      "type": "info",
    },
    Object {
      "data": Object {
        "dependencies": 100,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 100,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "line-chart": Array [
    Object {
      "data": Object {
        "dependencies": 112,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 112,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "parallel-plot": Array [
    Object {
      "data": Object {
        "dependencies": 111,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 111,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "pie-chart": Array [
    Object {
      "data": Object {
        "dependencies": 109,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 109,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "scatter-plot": Array [
    Object {
      "data": Object {
        "dependencies": 109,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 109,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "stacked-bar-chart": Array [
    Object {
      "data": Object {
        "dependencies": 109,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 109,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "types": Array [
    Object {
      "data": Object {
        "dependencies": 1,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 1,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "utils": Array [
    Object {
      "data": Object {
        "dependencies": 194,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 194,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "utils-dev": Array [
    Object {
      "data": Object {
        "dependencies": 203,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 203,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "visa-chart-components": Array [
    Object {
      "data": Object {
        "dependencies": 2357,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 2357,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
  "world-map": Array [
    Object {
      "data": Object {
        "dependencies": 120,
        "devDependencies": 0,
        "optionalDependencies": 0,
        "totalDependencies": 120,
        "vulnerabilities": Object {
          "critical": 0,
          "high": 0,
          "info": 0,
          "low": 0,
          "moderate": 0,
        },
      },
      "type": "auditSummary",
    },
  ],
}
`;
